### PR TITLE
QA improvements

### PR DESCRIPTION
**Pull request by ArdonToonstra**
Hi Pieter,

I try to mitigate two types of errors by the QA tooling:

* "sd-pg-02: 'If mapping.map exists and the mapping is not implicit, short should exist'" for the nl-core profiles. I did this by moving the constraints from the general SD to the SD for zib-profiles.
* "StructureDefinition.purpose: minimum required = 1, but only found 0" for extension that are created in the nl-core folder. This is not solved yet. I think this should be adjusted in the validate yml file or something. Could you have a look?

### CODE REVIEW

**Comment by ArdonToonstra** [comment]
Added another goal: to remove an error if an extension does not contain a reference to a base FHIR base resource. Just tested the new FHIRpath expression in MedicationProfiles branch with succes.

**Comment by pieter-edelman-nictiz** [comment]
There are currently two kinds of extensions: extensions that represent a zib concept (they contain a dot) and other extensions. For the zib concept extensions, a StructureDefinition exists, while the other extensions are checked using the basic SD. These rules apply as well to the extensions in the nl-core folder. What rules what be suitable?
- In zib folder and containing a dot: zib concept with custom SD
- In nl-core folder and containing a dot: use case concept, another custom SD
- Other extensions: the basic profiling guidelines SD

File: qa/ProfilingGuidelinesR4-StructureDefinitions-Zib.xml [hunk 1]
```diff
@@ -69,19 +69,36 @@
...
       </constraint>
-    </element>
-    <element id="StructureDefinition.differential.element.type">
-      <path value="StructureDefinition.differential.element.type" />
       <constraint>
         <key value="sd-zpg-03" />
         <severity value="error" />
-        <human value="References to other resource profiles should be added next to the HL7  base references." />
-        <expression value="code = 'Reference' implies targetProfile.where($this.startsWith('http://hl7.org/fhir/StructureDefinition/')).exists()" />
+        <human value="References to other resource profiles should be added next to the HL7  base references. This does not apply for extensions." />
```
Comment (pieter-edelman-nictiz) [inline qa/ProfilingGuidelinesR4-StructureDefinitions-Zib.xml:100]: This is the wrong approach. The check can simply be moved to the StructureDefinitions-Zib-Profiles profile.

Comment (ArdonToonstra) [reply qa/ProfilingGuidelinesR4-StructureDefinitions-Zib.xml]: You are right. Adjusted per your feedback.

**Comment by ArdonToonstra** [comment]
@pieter-edelman-nictiz ,
> - In zib folder and containing a dot: zib concept with custom SD
> - In nl-core folder and containing a dot: use case concept, another custom SD
> - Other extensions: the basic profiling guidelines SD

Seems ok to me! 
I made a small adjustment based on your feedback. All seems to work. Ready to merge?

**Pull request merged**